### DISCUSSION TITLE
Delete fields that do not contain a value in property grid

### DISCUSSION THREAD

**Discussion by Mohammed-Benotmane** [iTwin/itwinjs-core#3632] [Q&A]
I am trying to remove empty fields from the property table, Example:

![empty](https://user-images.githubusercontent.com/39698080/168847115-dd2e4f1d-f1f2-495a-8295-17c8408380b4.png)

I am using code from this sandbox: https://www.itwinjs.org/sandboxes/iTwinPlatform/Property%20Formatting

In the PropertyGridComponent.tsx I set ```includeFieldsWithNoValues``` to ```false```:

```typescript
useEffect(() => {
    if (iModelConnection && selectionContext) {
      const provider = PropertyFormattingApi.createPropertyDataProvider(new KeySet(selectionContext.getSelection()), iModelConnection, customized);
      provider.isNestedPropertyCategoryGroupingEnabled = true;
      provider.includeFieldsWithNoValues = false;
      provider.keys = new KeySet(selectionContext.getSelection());
      setSelectionState({ isOverLimit: provider.keys.size > MAX_ELEMENTS, size: provider.keys.size });
      setDataProvider(provider);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [iModelConnection, selectionContext?.getSelection, customized]);
```
For Navisworks & Revit models I had this Error:

![Capture](https://user-images.githubusercontent.com/39698080/168815544-a0e7f1d7-9f98-4da8-9213-cb7dc37259d7.PNG)

For Synchro models it works but the details displayed are not descriptive, example:

![nch](https://user-images.githubusercontent.com/39698080/168854465-fda13e99-b47a-4003-81a1-d0e0629f5c05.PNG)

**Comment by grigasp** [comment]
Thanks for reporting this. I filed #3637 regarding the assertion failure in Navisworks & Revit iModels.

> For Synchro models it works but the details displayed are not descriptive

Can you give more details on this? Which part exactly is not descriptive? Is there information missing or wrong?

**Reply by Mohammed-Benotmane** [reply]
The details for the iModels that are imported in Synchro from Revit or Navisworks are missing, I think Synchro doesn't take the information correctly.

All details are transformed to the same information like in the image i sent.

**Reply by grigasp** [reply]
@abeesh and @jffmarker

**Reply by jffmarker** [reply]
So your concern is that the Code value is empty (i.e. not getting converted), but you only see this happen when using Synchro? Would you be willing to share the RVT file so we can reproduce? If you need to share it privately, please start a [service request](https://apps.bentley.com/srmanager/).
